rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend

```diff
@@ -110,39 +148,87 @@ class TestStepSelector {
 //		// startup complete ensures the index to be populated
 //		Display.getDefault.syncExec[populateViewIfEmpty] // is only run, if view is created on startup
 //	}
-
 	@Focus
 	def void setFocus() {
 		viewer.control.setFocus
 	}
 
 	@Inject
 	@Optional
+	def void updateView(@EventTopic(SELECTOR_TOPIC_UPDATE) Object data) {
+		logger.debug("updateView.")
+		Display.getDefault.syncExec[updateView(true)]
+	}
+
+	@Inject
+	@Optional
 	def void refreshView(@EventTopic(SELECTOR_TOPIC_REFRESH) Object data) {
-		populated = false
-		logger.debug("Refreshing view.")
-		Display.getDefault.syncExec[populateViewIfEmpty]
+		logger.debug("refreshView.")
+		Display.getDefault.syncExec[updateView(false)]
 	}
 
-	// make sure that the index is populated before calling this method
-	// and that it is run in the swt thread!
-	private def populateViewIfEmpty() {
-		if (populated) {
+	private def updateView(boolean update) {
```

Comment:
There are three paths through this method:

1. Initialize the view if not done so far (regardless of the value of `update`)
2. Update the view if `update == true`
3. Don't do anything if the view is already initialized and `update == false`

This confuses ma as the method is called `updateView` so I would expect that it only performs step 2.
Can we split this behavior into two methods and call none of them in case of 3.?